src/docgen/schema.ts

```diff
@@ -0,0 +1,428 @@
+/**
+ * Describes any kind of type. This could be a primitive, a user-defined type
+ * (like `Bucket`), or a composition of types (like `Map<string, Bucket>[]`).
+ */
+export interface TypeSchema {
+  /**
+   * The language-specific name of the type. May contain "%" placeholder
+   * values to indicate references to types defined in the "types" field.
+   *
+   * @example "string"
+   * @example "%"
+   * @example "typing.List[%]"
+   * @example "Map<%, %>"
+   */
+  readonly name: string;
+
+  /**
+   * Types referenced within the "name" field.
+   */
+  readonly types?: (TypeSchema | JsiiEntity)[];
+}
+
+/**
+ * Describes a single "entity" in the jsii type system. This may be a type,
+ * but it could also be a property, method, parameter, enum member, etc.
+ */
+export interface JsiiEntity extends AssemblyMetadataSchema {
+  /**
+   * The language-specific type FQN.
+   */
+  readonly fqn: string;
```

Comment:
How will we get to a simple `displayName` here? Is that always the last `.`-separated part of the `fqn` ?

Because that might be incorrect in languages in which the `fqn` is not `.`-separated, or in which we want to have more than one component part of the displayName.

Better to make it a separate field I think.

Reply:
Hmm, yeah I agree that could be a risky assumption to make. I'll add in `displayName` 👍